components/nifosg/nifloader.cpp

```diff
@@ -364,6 +366,67 @@ namespace NifOsg
 
         void handleEffect(const Nif::Node* nifNode, osg::Node* node, Resource::ImageManager* imageManager)
         {
+            if (nifNode->recType == Nif::RC_NiLight)
+            {
+                const Nif::NiLight* nilight = static_cast<const Nif::NiLight*>(nifNode);
+                //TOFIX
+                float radius = std::max(1000.0f, nilight->boundXYZ.length());
+                //store osg::LightSource and convert it at instanciation to SceneUtil::LightSource
+                osg::ref_ptr<osg::LightSource> lightSource (new osg::LightSource);
+                //record this arbitrary radius for instanciation setup
+                lightSource->setUserData(new osg::FloatValueObject("radius",radius));
+                osg::ref_ptr<osg::Light> light (new osg::Light);
+
+                const Nif::NiPointLight* nipointlight = dynamic_cast<const Nif::NiPointLight*>(nifNode);
```

Comment:
Bad. Very bad. Terrible. We don't need dynamic casts to know the type of a NIF record. Just add RC_NiPointLight, RC_NiSpotLight etc. record types, update makeFactory accordingly and use them here.

Reply:
AFAIK Nif::NiPointLight has recType==RC_Nilight so the dynamicast is mandatory

Reply:
Please reread what I said. No, it's by no means mandatory.

Reply:
...so tell me what you'd put...

Reply:
ok i'll add others RC_types

Reply:
Just modify entries in the niffile.cpp, so different records will use different types:

```
    newFactory.insert(makeEntry("NiAmbientLight",             &construct <NiLight>                     , RC_NiAmbientLight                       ));
    newFactory.insert(makeEntry("NiDirectionalLight",         &construct <NiLight>                     , RC_NiDirectionalLight                       ));
    newFactory.insert(makeEntry("NiPointLight",               &construct <NiPointLight>                , RC_NiPointLight                       ));
    newFactory.insert(makeEntry("NiSpotLight",                &construct <NiSpotLight>                 , RC_NiSpotLight                       ));```

Reply:
done